Feat/schedule

99jaeyoung opened the pull request:
feat(schedule) : Add Entity, Repository, Add gradle dependencies

Add City, Country, Destination, Route, Schedule, DetailSchedule Entity, Repository

99jaeyoung committed:
feat(schedule) : Add Entity, Repository
feat(schedule) : Add Entity, Repository, Add gradle dependencies
feat(main-schedule) : Add DTO, revise Entity, Repository
- Change "type" field to Enum, Add Enum files, EnumConverter
- Add load googleMap API
- Add Schedule Data to frontend
- Add front frame for schedule page

Johnywhisky commented:
toEntity, toDto 등 다 만들어지면 머지 하는걸로 :)

O0oO0Oo commented on src/main/java/com/scit/letsleave/domain/destination/entity/DestinationEntity.java:
```diff
@@ -0,0 +1,90 @@
+    private CityEntity city;
+
+    @Column(name = "type", nullable = false)
+    private String type;
```
@Johnywhisky @99jaeyoung 
이 부분은 `@Enumerate` 를 사용하는게 좋을것 같습니다.

기존의 스트링 타입으로 관리하면, 존재하지 않는 밸류 검사를 위한 로직을 따로 추가해야하는데, enum 으로 값들을 관리한다면, 로직 추가 없이 enum 에 존재하지 않는 밸류로 요청이 들어왔을때, db 단까지 갈 필요없이 컨트롤러에서 잘못된 요청이라는 응답을 바로 줄수가 있습니다.

```java
public enum DestinationType {
      RESTAURANT, ACCOMMODATION, ....
}

public class DestinationEntity {
      ....
      @Enumerated(EnumType.STRING)
      private DestinationType destinationType
      ....
 }
```

99jaeyoung replied:
확인했습니다! 처리 방식이 약간 달라서 TODO부분으로 처리해놓았습니다. 확인해주시면 감사하겠습니다.

O0oO0Oo reviewed (approved):
👍

Johnywhisky commented on src/main/java/com/scit/letsleave/domain/destination/controller/DestinationController.java:
```diff
@@ -0,0 +1,30 @@
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/destination")
```
rest api에 맞게 destination -> destinations로 수정 하면 좋을 거 같아요

Johnywhisky commented on src/main/java/com/scit/letsleave/domain/destination/dto/CityDTO.java:
```diff
@@ -0,0 +1,49 @@
+    private LocalDateTime createdAt;
+    private LocalDateTime updatedAt;
+
+    // ManyToOne 관계: cities.country_id → countries.id
```
해당 주석은 db로도 확인 가능하니 삭제해도 좋을 거 같아요

Johnywhisky commented:
ToDo: DestinationScheduleService에서 Dto를 하기의 패턴을 채용해 깔끔하게 수정하는 것도 고려
``` java
@Component
public class UserDTOMapper {
    public UserResponseDTO toUserResponseDTO(User user) {
        return new UserResponseDTO(
                user.getId(),
                user.getUsername(),
                user.getEmail()
        );
    }

    public User fromUserRegistrationDTO(UserRegistrationDTO userRegistrationDTO) {
        return User.builder()
                .username(userRegistrationDTO.username())
                .email(userRegistrationDTO.email())
                .password(userRegistrationDTO.password())
                .build();
    }
}
```

99jaeyoung committed:
feat(main-schedule) : modify file style to RESTAPI, Add coordinate field, Add footer frame, Add schedule check

- Add gradle dependencies
- Add coordinate field, mapping lat, lng
- Add schedule exist function
- Add footer frame

Johnywhisky committed:
Merge branch 'main' into feat/schedule
